### PR TITLE
chore: synchronize with Python templates

### PR DESCRIPTION
**Pull request by tswast**
Please review https://github.com/googleapis/python-bigquery-magics/pull/8 first! This PR depends on changes made in that PR.

Only 1 file here is hand-written:

* `owlbot.py`

### CODE REVIEW

**Comment by tswast** [comment]
Might be a good idea to get `setup.py` and unit tests working before continuing with this PR.

**Comment by tswast** [comment]
Marking as `do not merge` to wait for #8 to merge first.

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -22,7 +22,7 @@ In order to add a feature:
   documentation.
 
 - The feature must work fully on the following CPython versions:
-  3.7, 3.8, 3.9, 3.10, 3.11 and 3.12 on both UNIX and Windows.
+  3.7, 3.8, 3.11 and 3.12 on both UNIX and Windows.
```
Comment (chalmerlowe) [inline CONTRIBUTING.rst:25]: Curious: why the de-emphasis on versions 3.9 and 3.10?

**Comment by chalmerlowe** [comment]
> Please review #8 first! This PR depends on changes made in that PR.
> 
> Only 1 file here is hand-written:
> 
> * `owlbot.py`

I don't see `owlbot.py` in this PR.

**Review by chelsea-lin — APPROVED** [review]
LGTM overall

File: CONTRIBUTING.rst (hunk 1 above)
Comment (tswast) [reply CONTRIBUTING.rst:25]: I think this is taking the unit test versions from owlbot. We do want it to work on these, but I wonder if it's worth testing against so many versions?

**Comment by tswast** [comment]
> I don't see owlbot.py in this PR.

Oops. Merged that in https://github.com/googleapis/python-bigquery-magics/pull/8

The owlbot lock files actually activates it.

**Pull request merged**